src/Raven.Server/Documents/Sharding/Subscriptions/ShardedDocumentsDatabaseSubscriptionProcessor.cs

```diff
@@ -47,32 +55,49 @@ protected override bool ShouldSend(Document item, out string reason, out Excepti
         exception = null;
         result = item;
 
-        if (Fetcher.FetchingFrom == SubscriptionFetcher.FetchingOrigin.Resend)
+        if (IsUnderActiveMigration(item.Id, _sharding, _allocator, _database.ShardNumber, Fetcher.FetchingFrom, out reason, out bool isActiveMigration))
         {
-            var bucket = ShardHelper.GetBucketFor(_sharding, _allocator, item.Id);
-           
-            if (_sharding.BucketMigrations.TryGetValue(bucket, out var migration))
+            IsActiveMigration = isActiveMigration;
```

Comment:
I think it is better to pack all `out` arguments into some struct, also this is per item, so should be there?

Reply:
done